app/models/message.rb

```diff
@@ -195,7 +195,7 @@ def determine_user_validity
   def channel_permission
     errors.add(:base, "Must be part of channel.") if chat_channel_id.blank?
 
-    channel = ChatChannel.find(chat_channel_id)
+    channel = chat_channel || ChatChannel.find(chat_channel_id)
```

Comment:
Snuck in this little optimization to avoid redundantly looking up the chat_channel for existing messages